### PR TITLE
Make ResponseBodyNotFoundException internal

### PR DESCRIPTION
**Pull request by annelo-msft**
Per discussion with @KrzysztofCwalina and @pakrym, making this exception not a publicly visible type.

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

**Review by KrzysztofCwalina — APPROVED** [review]

**Comment by annelo-msft** [comment]
Fixed build break where GetBlobsOptions was still around.  @JoshLove-msft please check my work.

**Comment by JoshLove-msft** [comment]
Didn't realize EventHubs was using that API. Looks good - thanks for fixing!

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -666,11 +666,18 @@ public async Task GetIfChangedSettingUnmodified()
                 // Test
                 Response<ConfigurationSetting> response = await service.GetAsync(setting, onlyIfChanged: true).ConfigureAwait(false);
                 Assert.AreEqual(304, response.GetRawResponse().Status);
-                var exception = Assert.Throws<ResponseBodyNotFoundException>(() =>
+
+                bool throws = false;
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: https://github.com/nunit/docs/wiki/Assert.Catch

**Review by jsquire — COMMENTED** [review]
Late to the party, but the Event Hubs change looks good to me.

**Pull request merged**